### PR TITLE
Display "Invalid number" error on invalid number of days

### PR DESCRIPTION
**Pull request by PrestaEdit**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Display "Invalid number" error on invalid number of days.
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes PrestaShop/Prestashop#29933.
| How to test?  | See issue.

### CODE REVIEW

File: ps_newproducts.php [hunk 1]
```diff
@@ -110,12 +110,13 @@ public function getContent()
...
         if (Tools::isSubmit('submitBlockNewProducts')) {
             $productNbr = Tools::getValue('NEW_PRODUCTS_NBR');
+            $productNbDays = Tools::getValue('PS_NB_DAYS_NEW_PRODUCT');
 
             if (!$productNbr || empty($productNbr)) {
                 $output .= $this->displayError(
                     $this->trans('Please complete the "products to display" field.', [], 'Modules.Newproducts.Admin')
                 );
-            } elseif (0 === (int) $productNbr) {
+            } elseif (0 === (int) $productNbr || 0 === (int)$productNbDay) {
```
Comment (PrestaEdit) [inline ps_newproducts.php:119]:
```suggestion
            } elseif (0 === (int) $productNbr || 0 === (int)$productNbDays) {
```

File: ps_newproducts.php [hunk 2]
```diff
@@ -110,12 +110,13 @@ public function getContent()
...
         if (Tools::isSubmit('submitBlockNewProducts')) {
             $productNbr = Tools::getValue('NEW_PRODUCTS_NBR');
+            $productNbDays = Tools::getValue('PS_NB_DAYS_NEW_PRODUCT');
 
             if (!$productNbr || empty($productNbr)) {
                 $output .= $this->displayError(
                     $this->trans('Please complete the "products to display" field.', [], 'Modules.Newproducts.Admin')
                 );
-            } elseif (0 === (int) $productNbr) {
+            } elseif (0 === (int) $productNbr || 0 === (int)$productNbDays) {
```
Comment (matks) [inline ps_newproducts.php:119]:
```suggestion
            } elseif (0 === (int) $productNbr || 0 === (int) $productNbDays) {
```

Code is OK but PHP-CS-Fixer wants this little white space 😅

**Review by mflasquin — APPROVED** [review]

**Review by lartist — APPROVED** [review]

**Review by nicosomb — APPROVED** [review]

**Review by okom3pom — APPROVED** [review]

**Review by FabienPapet — APPROVED** [review]

**Comment by paulnoelcholot** [comment]
Hello @PrestaEdit,

I tested the PR in 8.0 and I still reproduce the bug. It is always possible to introduce letters and special characters in the input, as you can see in the video.

https://watch.screencastify.com/v/9W9wUf6yrfP3pA5QvQqD

**Comment by PrestaEdit** [comment]
Hi,

Could you verify that you have the change into your test, maybe ?

**Comment by paulnoelcholot** [comment]
My apologies, indeed the problem came from my environment. Once this is corrected, the PR works well.

Thanks for your PR. The issue is well fixed and I found no regression.

**Review by paulnoelcholot — APPROVED** [review]

**Comment by jolelievre** [comment]
Thanks @PrestaEdit and @paulnoelcholot

**Pull request merged**